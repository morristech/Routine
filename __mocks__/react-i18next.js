/*
 * This file is part of the Routine project.
 *
 * (c) Yasser Ameur El Idrissi <getspookydev@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

'use strict';

jest.mock('react-i18next', () => ({
  withNamespaces: () => (Component) => {
    Component.defaultProps = {
      ...Component.defaultProps,
      t: (key) => key,
    };
    return Component;
  },
}));
