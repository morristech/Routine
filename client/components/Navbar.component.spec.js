/*
 * This file is part of the Routine project.
 *
 * (c) Yasser Ameur El Idrissi <getspookydev@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import React from 'react';
import ReactDom from 'react-dom';
import '../../__mocks__/react-i18next';
import NavbarComponent from './Navbar.component';

let wrapper = null;

beforeEach(function () {
  wrapper = <NavbarComponent logoSrc="https://example/com/image.png" />;
});

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(wrapper, div);
  ReactDom.unmountComponentAtNode(div);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
