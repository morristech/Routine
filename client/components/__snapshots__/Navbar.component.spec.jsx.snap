// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<NavbarComponent
  logoSrc="https://example/com/image.png"
  t={[Function]}
/>
`;
